src/App.js

```diff
@@ -14,11 +15,99 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    allGoods: goodsFromServer,
+    selectedGoods: [],
+  }
+
+  clearAll = () => {
+    this.setState({
+      selectedGoods: [],
+    });
+  }
+
+  addSelected = (good) => {
+    this.setState(prevState => ({
+      selectedGoods: [...prevState.selectedGoods, good],
+    }));
+  }
+
+  removeSelected = (good) => {
+    this.setState((prevState) => {
+      const i = prevState.selectedGoods.indexOf(good);
+
+      prevState.selectedGoods.splice(i, 1);
+
+      return (
+        {
+          selectedGoods: prevState.selectedGoods,
+        }
+      );
+    });
+  }
+
+  render() {
+    const { allGoods, selectedGoods } = this.state;
+
+    return (
+      <div className="App">
+        <h1 className="App__header">
+          Selected goods:
+          {' '}
+          {selectedGoods.length}
+        </h1>
+
+        {selectedGoods.map(good => (
+          <span
+            key={good}
+            className="App__selected"
+          >
+            {`${good}, `}
+          </span>
+        ))}
+        <br />
+
+        {selectedGoods.length < 10 && selectedGoods.length !== 0
```

Comment:
Why would we hide 'Clear all' button when all goods are selected? If I select all the goods and want to deselect them in one click what should I do? In my opinion, check `selectedGoods.length !== 0` would be enough

Reply:
Agree. I misunderstood indtructions